File: aoc2022/day1/day1-2.m
#import <Foundation/Foundation.h>

NSInteger intSort(id num1, id num2, void *context)
{
    int v1 = [num1 intValue];
    int v2 = [num2 intValue];
    if (v1 > v2)
        return NSOrderedAscending;
    else if (v1 < v2)
        return NSOrderedDescending;
    else
        return NSOrderedSame;
}

int main() {
    NSAutoreleasePool *pool = [[NSAutoreleasePool alloc] init];
    NSString* filePath = @"input.txt";
    NSString* fileContent = [NSString stringWithContentsOfFile:@"../input.txt"
                            encoding:NSUTF8StringEncoding error:nil];
    NSArray* lines = [fileContent componentsSeparatedByString:(NSString *)@"\n"];
    NSMutableArray* scores = [[NSMutableArray alloc] init];
    int runningTotal = 0;
    for (NSString* line in lines) {
        if ([line length] > 0) {
            int currentNum = [line intValue];
            runningTotal = runningTotal + currentNum; 
            NSLog(@"runningTotal: %d", runningTotal); 
        } else {
            NSLog(@"*** new elf");
            [scores addObject:[NSNumber numberWithInteger:runningTotal]];
            runningTotal = 0;
        }
    }
    NSLog(@"scores: %@", scores);
    // NSArray* topScores = [scores sortedArrayUsingSelector:@selector(integerValue)];
    NSArray* topScores; 
    topScores = [scores sortedArrayUsingFunction:intSort context:NULL];

    NSLog(@"topScores: %@", topScores);
    int total = [topScores[0] intValue] + [topScores[1] intValue] + [topScores[2] intValue];
    NSLog(@"total: %d", total);

    [pool drain];
    return 0;
}

File: aoc2022/day1/day1-2.swift
#!/usr/bin/env swift

import Foundation

do {
    let fileContent = try NSString(contentsOfFile: "input.txt", encoding: String.Encoding.utf8.rawValue)
    let lines = fileContent.components(separatedBy: "\n")
    var runningTotal = 0
    var totals: [Int] = []
    for line in lines {
        if line.count < 1 {
            totals.append(runningTotal)
            runningTotal = 0
        } else {
            let currentVal = Int(line)
            runningTotal = runningTotal + currentVal!
        }
    }
    totals.sort()
    totals.reverse()
    print("totals: \(totals)")
    let total = totals[0] + totals[1] + totals[2]
    print("total: \(total)")
} catch {
    print(error)
}